src/bundle.c

```diff
@@ -29,15 +30,33 @@ static gboolean mksquashfs(const gchar *bundlename, const gchar *contentdir, GEr
 		goto out;
 	}
 
-	sproc = r_subprocess_new(G_SUBPROCESS_FLAGS_STDOUT_SILENCE,
-			&ierror, "mksquashfs",
-			contentdir,
-			bundlename,
-			"-all-root",
-			"-noappend",
-			"-no-progress",
-			"-no-xattrs",
-			NULL);
+	g_ptr_array_add(args, g_strdup("mksquashfs"));
+	g_ptr_array_add(args, g_strdup(contentdir));
+	g_ptr_array_add(args, g_strdup(bundlename));
+	g_ptr_array_add(args, g_strdup("-all-root"));
+	g_ptr_array_add(args, g_strdup("-noappend"));
+	g_ptr_array_add(args, g_strdup("-no-progress"));
+	g_ptr_array_add(args, g_strdup("-no-xattrs"));
+
+
+	if (r_context()->mksquashfs_args != NULL) {
+		g_auto(GStrv) mksquashfs_argvp = NULL;
+		res = g_shell_parse_argv(r_context()->mksquashfs_args, NULL, &mksquashfs_argvp, &ierror);
```

Comment:
Oups... Consider the case we have `mkquashfs_args` set and they are parsed correctly.
This sets `res` to `TRUE`.
Now, in line 59, the `r_subprocess_newv()` call returns an error for whatever reason.
This will not modify `res` (as it sets `sproc` only) but lead us directly to the `out` jump label (with ierror set)
At the `out` label, we `return res` from the function.

At this point, the execution failed, `error` is set, but the function returns 'TRUE` that suggests a successful run!

Reply:
Adding `res = FALSE;` in the error path of `g_subprocess_newv()` should to the trick. We have similar cases in other functions in the same module.

Reply:
Yep, makes sense. Updated :)

Reply:
@ejoerns: that was a good catch!